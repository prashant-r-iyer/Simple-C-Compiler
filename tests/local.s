	.text

.globl main
main:
	# Save Frame pointer
	pushq %rbp
	movq %rsp, %rbp
	subq $256, %rsp
	pushq %rbx
	pushq %rbx
	pushq %r10
	pushq %r13
	pushq %r14
	pushq %r15
	pushq %r11
	pushq %r9

	# push 6
	movq $6, %rbx
	movq %rbx, -8(%rbp)
	movq $string0, %rbx
	movq -8(%rbp), %r10
	# func=printf nargs=2

	# Move values from reg stack to reg args
	movq %r10, %rsi
	movq %rbx, %rdi
	movl $0, %eax
	call printf
	movq %rax, %rbx

	# Restore registers
	popq %r9
	popq %r11
	popq %r15
	popq %r14
	popq %r13
	popq %r10
	popq %rbx
	popq %rbx
	addq $256, %rsp

	leave
	ret

string0:
	.string "l=%d\n"

