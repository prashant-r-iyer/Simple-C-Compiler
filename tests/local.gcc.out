local.c: In function ‘main’:
local.c:7:9: warning: implicit declaration of function ‘printf’ [-Wimplicit-function-declaration]
    7 |         printf("l=%d\n", l);
      |         ^~~~~~
local.c:1:1: note: include ‘<stdio.h>’ or provide a declaration of ‘printf’
  +++ |+#include <stdio.h>
    1 | 
local.c:7:9: warning: incompatible implicit declaration of built-in function ‘printf’ [-Wbuiltin-declaration-mismatch]
    7 |         printf("l=%d\n", l);
      |         ^~~~~~
local.c:7:9: note: include ‘<stdio.h>’ or provide a declaration of ‘printf’
local.c:7:20: warning: format ‘%d’ expects argument of type ‘int’, but argument 2 has type ‘long int’ [-Wformat=]
    7 |         printf("l=%d\n", l);
      |                   ~^     ~
      |                    |     |
      |                    int   long int
      |                   %ld
